app/controllers/v3/service_brokers_controller.rb

```diff
@@ -75,9 +75,10 @@ def update
 
     if service_broker.space_guid
       space = service_broker.space
-      unprocessable_space! unless space && permission_queryer.can_read_from_space?(space.guid, space.organization_guid)
+      broker_not_found! unless space && permission_queryer.can_read_from_space?(space.guid, space.organization_guid)
```

Comment:
I think scenario 1c says this should be a 422 error, and this would appear to change it to a 404